src/ORM/Association.php

```diff
@@ -886,7 +887,7 @@ public function setFinder($finder)
      *
      * @deprecated 3.4.0 Use setFinder()/getFinder() instead.
      * @param string|null $finder the finder name to use
-     * @return string
+     * @return string|array
```

Comment:
Return type of `getFinder()` method should also be updated.

Reply:
pushed https://github.com/cakephp/cakephp/pull/12770/commits/1a46ad998e12bd47314b6bf1fba1e6a4e202c2ff 😄